awx_collection/plugins/module_utils/ansible_tower.py

```diff
@@ -46,8 +51,30 @@
 from ansible.module_utils.basic import AnsibleModule, missing_required_lib
 
 
+def make_request(request_handler, tower_url):
+    '''
+    Makes the request to given URL, handles errors, returns JSON
+    '''
+    try:
+        response = request_handler.get(tower_url)
+    except (ConnectionError, urllib_error.URLError, socket.error, httplib.HTTPException) as e:
+        n_error_msg = 'Connection to remote host failed: {err}'.format(err=to_native(e))
+        # If Tower gives a readable error message, display that message to the user.
+        if callable(getattr(e, 'read', None)):
+            n_error_msg += ' with message: {err_msg}'.format(err_msg=to_native(e.read()))
+        raise CollectionsParserError(n_error_msg)
+
+    # Attempt to parse JSON.
+    try:
+        return json.loads(response.read())
+    except (ValueError, TypeError) as e:
+        # If the JSON parse fails, print the ValueError
+        raise CollectionsParserError('Failed to parse json from host: {err}'.format(err=to_native(e)))
```

Comment:
these will still surface as a traceback, but the prior behavior (and what we probably want) is to error with just this message. This will probably require catching and re-throwing as the old `AnsibleParserError`, but only in the inventory plugin.